### PR TITLE
migrate to j hipster architecture

### PR DESCRIPTION
**Pull request by stibos**
What's done:
- security
- biggest part of refactoring
- public part (systems/subscriptions)
- system management in admin area
- message readonly in admin area
- new grunt build task, without alv-ch-build

What's still to do:
- message creation / management (i'll also implement the message-state feature, but i miss some information)
- migration to the latest alv-ch-ng version released yesterday
- cleanup
- testcases

### CODE REVIEW

**Comment by nidDrBiglr** [comment]
More information about MessageState management: 


Behavior / management of SystemMessages designed with & by the PO Mr B.D.:

    MaintenanceWindows
        States in use: Planned (P), in Progress (iP), Resolved (R), Cancelled (C)
        State changes:
            P > iP: time event (start dateTime)
            iP > R: time event (end dateTime) || triggered by admin user (change end dateTime to event time)
            R > C: invalid state change, one can't cancel if its already resolved
            P > C: triggered by admin user
            iP > C: invalid state change, one can't cancel if its already in progress
            all other, not explicitely mentioned state changes are illegal ones
        Reminder Mails:
            MaintenanceWindows may trigger reminder e-mails which will be sent a certain amount of time before the window starts. For this purpose, an additional checkbox ("send reminder") and time dropdown must be placed in the maintenanceWindow form
        Resolved Mails: When a maintenanceWindow reaches the state 'resolved', an e-mail may be sent to subscribers. If this should happen, the dedicated checkbox (is not there yet) must be checked by the admin user on message creation time.
        Mail sending in general: Immediately after a successful message creation an if reminder or resolved-mail is enabled

    Incidents
        States in use: Investigating (I), Resolving (R), Closed (C)
        allowed state changes:
            I > R
            I > C
            R > C
        No time events, all state changes are manually done be an admin user
        Always use timestamp of manual event when setting dateTime
        Mail sending: send mail on creation AND when state changes to closed

If that is what you're looking for :smile:

**Comment by stibos** [comment]
@nidDrBiglr yes indeed, that info I was missing. Thanks buddy!

**Comment by nidDrBiglr** [comment]
@stibos you are very welcome my dear sire :laughing:

File: .bowerrc [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  "directory" : "components"
+  "directory" : "src/bower_components"
```
Comment (gildegoma) [inline .bowerrc]: really? If yes, please update `.gitignore` file.

File: .gitignore [hunk 2]
```diff
@@ -12,4 +12,4 @@ private/**/*
 test/reports/**/*
 test/coverage/**/*
 _SpecRunner.html
-build/dist/alv-ch-sysinfos.common.config.js
+build/dist/sysinfos.config.js
```
Comment (gildegoma) [inline .gitignore]: Also ignore `src/styles/themes/flick` ?

**Comment by gildegoma** [comment]
@stibos First feedbacks:

* `dev` partially works (the system names are not displayed for example)
* `dist` doesn't work because `angular` is not loaded due to a syntax error in `application.min.js:25` :  `Uncaught SyntaxError: Unexpected token ILLEGAL`. Any idea?

**Comment by gildegoma** [comment]
Precisions about:

> dev partially works (the system names are not displayed for example)

* it looks nice on Safari
* no system titles displayed in Chrome and Firefox

Note: Frutiger fonts are not activated at the moment.

**Comment by gildegoma** [comment]
> pending: migration to the latest alv-ch-ng version released yesterday

which version should be integrated then? (`bowser.json` is set to latest)

File: .bowerrc (hunk 1 above)
Comment (stibos) [inline .bowerrc]: really! .gitignore will be fixed

File: .gitignore (hunk 2 above)
Comment (stibos) [inline .gitignore]: yes, this is copied by the build task. no version control necessary!

**Comment by stibos** [comment]
i have look at the name and dist/angular things.

i do the migration to the next alv-ch-ng version this evening. So it's done when mergin'.

**Comment by stibos** [comment]
.gitIgnore is fixed and build process optimized.

i don't have any problems in displaying system names (chrome & ff on ubuntu). i've changed the default port of the node server script to 3000, that way you can run the application without browserSync and special config of the backend part. just run 'node server.js src' or 'node server.js dist'

**Comment by gildegoma** [comment]
Second wave of feedbacks:
* both `node server.js src` and `node server.js dist` now work fine for me (no more rendering troubles with the system names in Firefox and Chrome)
* but deploying `dist` application to an nginx web site fails with this **Syntax Error: illegal character** in `application.min.js:25`. I suppose these illegal characters are these ones:

begin:
image

end:
image

This is quite odd, because when I open `application.min.js` in a text editor:
* `$on("$routeChangeSuccess",function(){window.scrollTo(0,0)})})}();` corresponds to the end of the file, while
* `getFormGridContent:getFormGridContent,getFormGrid:getFormGrid` is **inserted twice**, the second time after the `��...��` characters.

image. I suppose it could be a bug/problem in file sharing/sync between host and vagrant vm, but it is almost certain that we have a deployment problem here.

Anyway, I merge this. Still to do:

* [ ] message creation / management
* [ ] clean up
* [ ] tests

**Pull request merged**